### ISSUE TITLE
show-on stop working after Mautic 5.2

### ISSUE THREAD

**Issue by kuzmany** [mautic/mautic#14402]
### Mautic Series

5.2.x series

### Mautic installed version

5.2

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.1

### What browsers are you seeing the problem on?

Chrome

### What happened?

Seems this PR https://github.com/mautic/mautic/pull/14221/ broke some states in data-show-on. 
Note:  works properly If use data-hide-on
Here is example for forms fields behaviour tab:

https://github.com/user-attachments/assets/efe4a21d-bf0a-4faf-88f7-a0e5200b5476

### How can we reproduce this issue?

Step 1:  Go to form
2: Create from
3: Create any field
4. On create new field go to behaviours tab
5. Enable Always visible
6. Disable Always visible
7. Repeat 
8. See like option Show when value exists disappeared

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by kuzmany** [comment]
@laurielim do you think you find some time to check it?

**Comment by laurielim** [comment]
Yes I'll look into it.

**Comment by laurielim** [comment]
Okay so this is a duplicate of #14365. I'm making a fix now for both issues

**Comment by laurielim** [comment]
Created PR https://github.com/mautic/mautic/pull/14416 for this, if you could please review it @kuzmany

**Comment by escopecz** [comment]
Fixed in https://github.com/mautic/mautic/pull/14416
